airflow/dag_processing/manager.py

```diff
@@ -298,74 +301,58 @@ class DagFileProcessorManager(LoggingMixin):
     """
 
-    def __init__(
-        self,
-        dag_directory: os.PathLike[str],
-        max_runs: int,
-        processor_timeout: timedelta,
-        signal_conn: MultiprocessingConnection | None = None,
-    ):
-        super().__init__()
-        # known files; this will be updated every `dag_dir_list_interval` and stuff added/removed accordingly
-        self._file_paths: list[str] = []
-        self._file_path_queue: deque[str] = deque()
-        self._max_runs = max_runs
-        # signal_conn is None for dag_processor_standalone mode.
-        self._direct_scheduler_conn = signal_conn
-        self._parsing_start_time: float | None = None
-        self._dag_directory = dag_directory
-        # Set the signal conn in to non-blocking mode, so that attempting to
-        # send when the buffer is full errors, rather than hangs for-ever
-        # attempting to send (this is to avoid deadlocks!)
-        if self._direct_scheduler_conn:
+    _dag_directory: os.PathLike[str]
+    max_runs: int
+    processor_timeout: float = attrs.field(factory=_config_int_factory("core", "dag_file_processor_timeout"))
+    selector: selectors.BaseSelector = attrs.field(factory=selectors.DefaultSelector)
+    _direct_scheduler_conn: MultiprocessingConnection | None = attrs.field(alias="signal_conn", default=None)
+
+    _parallelism: int = attrs.field(factory=_config_int_factory("scheduler", "parsing_processes"))
+
+    dag_dir_list_interval: int = attrs.field(
+        factory=_config_int_factory("scheduler", "dag_dir_list_interval")
+    )
+    parsing_cleanup_interval: float = attrs.field(
+        factory=_config_int_factory("scheduler", "parsing_cleanup_interval")
+    )
+    _file_process_interval: float = attrs.field(
+        factory=_config_int_factory("scheduler", "min_file_process_interval")
+    )
+    stale_dag_threshold: float = attrs.field(factory=_config_int_factory("scheduler", "stale_dag_threshold"))
+    last_dag_dir_refresh_time: float = attrs.field(default=0, init=False)
+
+    log: logging.Logger = log
+
+    _last_deactivate_stale_dags_time: float = attrs.field(default=0, init=False)
+    print_stats_interval: float = attrs.field(
+        factory=_config_int_factory("scheduler", "print_stats_interval")
+    )
+    last_stat_print_time: float = attrs.field(default=0, init=False)
+
+    _file_paths: list[str] = attrs.field(factory=list, init=False)
+    _file_path_queue: deque[str] = attrs.field(factory=deque, init=False)
+    _file_stats: dict[str, DagFileStat] = attrs.field(factory=lambda: defaultdict(DagFileStat), init=False)
+
+    _processors: dict[str, DagFileProcessorProcess] = attrs.field(factory=dict, init=False)
+
+    _parsing_start_time: float = attrs.field(init=False)
+    _num_run: int = attrs.field(default=0, init=False)
+
+    _callback_to_execute: dict[str, list[CallbackRequest]] = attrs.field(
+        factory=lambda: defaultdict(list), init=True
+    )
```

Comment:
```suggestion
    _file_paths: list[str] = attrs.field(factory=list, init=False)
    _file_path_queue: deque[str] = attrs.field(factory=deque, init=False)
    _file_stats: dict[str, DagFileStat] = attrs.field(factory=lambda: defaultdict(DagFileStat), init=False)

    _processors: dict[str, DagFileProcessorProcess] = attrs.field(factory=dict, init=False)

    _parsing_start_time: float = attrs.field(init=False)
    _num_run: int = attrs.field(default=0, init=False)

    _callback_to_execute: dict[str, list[CallbackRequest]] = attrs.field(
        factory=lambda: defaultdict(list), init=False
    )
```

I think

Reply:
Yeah it will probably show up in the `repr` otherwise